### PR TITLE
Feature/sign up

### PR DESCRIPTION
**Pull request by Omisandefunmi**
Added SignUp feature.
Kindly review.

### CODE REVIEW

File: Unsolo-rockets/src/main/java/com/interswitch/Unsolorockets/services/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+               .password(userDto.getPassword())
+               .phoneNumber(userDto.getPhoneNumber())
+               .build();
+
+        assignRole(userDto, user);
+        userRepository.save(user);
+        return SignUpResponse.builder()
+                .firstName(user.getFirstName())
+                .lastName(user.getLastName())
+                .email(user.getEmail())
```
Comment (jodugbose) [inline Unsolo-rockets/src/main/java/com/interswitch/Unsolorockets/services/impl/UserServiceImpl.java:46]: Can we find a way to confirm if the email is valid, I know there is regex on python, i don't know for Java, if there is a way to do it

Comment (Omisandefunmi) [reply Unsolo-rockets/src/main/java/com/interswitch/Unsolorockets/services/impl/UserServiceImpl.java:46]: Yes that will be implemented as well. This is not the complete work. This is done so as to let login begin.

Comment (jodugbose) [reply Unsolo-rockets/src/main/java/com/interswitch/Unsolorockets/services/impl/UserServiceImpl.java:46]: Alright, well done

**Review by jodugbose — APPROVED** [review]
Well done

**Review by AyeniAmanda — APPROVED** [review]

**Pull request merged**